src/factory.js

```diff
@@ -104,19 +105,19 @@ class Factory {
    * @returns {Promise<ControllerDaemon>}
    */
   async spawn (options = { }) {
+    const type = options.type || this.opts.type
     const opts = merge(
-      this.overrides[options.type || this.opts.type],
+      this.overrides[type],
+      // conditionally include ipfs based on which type of daemon we will spawn when none has been specifed
+      (type === 'js' || type === 'proc') ? {
+        ipfsModule: {
+          path: require.resolve('ipfs'),
+          ref: require('ipfs')
+        }
+      } : {},
       options
     )
 
-    // conditionally include ipfs based on which type of daemon we will spawn when none has been specifed
-    if ((opts.type === 'js' || opts.type === 'proc') && !opts.ipfsModule) {
```

Comment:
I was thinking this needed to be something more like:

```javascript
if ((opts.type === 'js' || opts.type === 'proc') && !opts.ipfsModule) {	
  opts.ipfsModule = {}
}

if (opts.ipfsModule && !opts.ipfsModule.path) {
  opts.ipfsModule.path = require.resolve('ipfs')
}

if (opts.ipfsModule && !opts.ipfsModule.ref) {
  opts.ipfsModule.path = require(opts.ipfsModule.path)
}
```

Reply:
with the current changes we don't need to protect it like that, its already safe 

https://codesandbox.io/s/clever-ride-kr3r1